spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CqlSessionFactoryBean.java

```diff
@@ -450,11 +451,20 @@ public void afterPropertiesSet() {
 
 		this.session = buildSession(sessionBuilder);
 
-		executeCql(getStartupScripts().stream(), this.session);
-		performSchemaAction();
+		try {
+			SchemaRefreshUtils.withDisabledSchema(this.session, () -> {
+				executeCql(getStartupScripts().stream(), this.session);
+				performSchemaAction();
+			});
+		} catch (RuntimeException e) {
```

Comment:
What's the reason for disabling schema refresh here? And why is there the need to declare and catch exceptions?

Reply:
I see. We can revisit exceptions in `performSchemaAction` to remove any unnecessary exceptions. 

> The goal here is to disable schema refresh before applying user-provided schema actions. Datastax4 waits for schema refresh after applying changes and if you're applying multiple it's really slow.

We should leave schema refresh up to the config. If schema refresh is disabled as per config, we don't need to disable it on our side.

Reply:
> We should leave schema refresh up to the config
for anyone looking at this thread in the future: my impression based on what you said is that we should rely only on schema being disabled via config in `performSchemaAction`, but I see you added `setSuspendLifecycleSchemaRefresh` and I'm a big fan!